src/repr/src/fixed_length.rs

```diff
@@ -0,0 +1,752 @@
+    /// can be provided. If the Row buffer is used, then the reference returned is to it.
+    /// Implementations are also free to place specific requirements on the given schema.
+    fn into_row(&'a self, row_buf: &'a mut Row, types: &[ColumnType]) -> &'a Row;
+}
+
+// Blanket identity implementation for Row.
+impl<'a> IntoRowByTypes<'a> for Row {
+    /// Performs a zero-copy reference forwarding without employing the `Row` buffer.
+    /// This implementation panics if non-empty `types` are provided.
+    fn into_row(&'a self, _row_buf: &'a mut Row, types: &[ColumnType]) -> &'a Row {
+        assert!(types.is_empty());
+        self
+    }
+}
+
+/// A helper trait to construct target values from input `Row` instances based on type
+/// information that only manifests at runtime (typically originating from inferred schemas).
+/// Typically, the target type will be of fixed-length without tags per column or `Row` itself.
+pub trait FromRowByTypes: Sized {
+    /// Obtains an instance of `Self` given an instance of `Row` and a schema provided
+    /// by `types`.
+    /// Implementations are free to place specific requirements on the given schema.
+    fn from_row(row: Row, types: &[ColumnType]) -> Self;
+
+    /// Obtains an instance of `Self' given an iterator of borrowed datums and a schema
+    /// provided by `types`.
+    /// Implementations are free to place specific requirements on the given schema.
+    fn from_datum_iter<'a, I, D>(datum_iter: I, types: &[ColumnType]) -> Self
+    where
+        I: IntoIterator<Item = D>,
+        D: Borrow<Datum<'a>>;
+}
+
+impl FromRowByTypes for Row {
+    /// Returns the provided row itself.
+    /// This implementation panics if non-empty `types` are provided.
+    fn from_row(row: Row, types: &[ColumnType]) -> Self {
+        assert!(types.is_empty());
+        row
+    }
+
+    /// Packs a `Row` from the given iterator of datums.
+    /// This implementation panics if non-empty `types` are provided.
+    fn from_datum_iter<'a, I, D>(datum_iter: I, types: &[ColumnType]) -> Self
+    where
+        I: IntoIterator<Item = D>,
+        D: Borrow<Datum<'a>>,
+    {
+        assert!(types.is_empty());
+        let mut row = Row::default();
+        row.packer().extend(datum_iter);
+        row
+    }
+}
+
+/// A representation of fixed-length values with N bytes, wherein one byte
+/// is reserved for nullability encoding.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct BytesN<const N: usize> {
+    #[serde(with = "BigArray")]
+    data: [u8; N],
+}
+
+// TODO(vmarcos): We may be able to simplify the implementation of Columnation
+// once frankmcsherry/columnation#18 lands.
+impl<const N: usize> Columnation for BytesN<N> {
+    type InnerRegion = CopyRegion<BytesN<N>>;
+}
+
+/// Helper function to read 8 bits as `bool`s out of a `u8`.
+fn u8_to_bool_slice(b: u8) -> [bool; 8] {
```

Comment:
That is absolutely correct. The same problem occurred in some functions I introduced in `row.rs`, so I fixed those. As for `u8_to_bool_slice`, I just removed this function for now. A variation of it will be useful when we encode up to 8 columns in a `BytesN` so that we can represent the nullability of these multiple columns in one byte. But there is no point in cluttering the code with this right now.